pkg/system/phase4_configuring.go

```diff
@@ -220,6 +222,24 @@ func (r *Reconciler) SetDesiredDeploymentEndpoint() {
 	}
 }
 
+// ReconcileHPAEndpoint reconcile the endpoint's HPS and report the configuration
+// back to the noobaa core
+func (r *Reconciler) ReconcileHPAEndpoint() error {
+	if err := r.ReconcileObject(r.HPAEndpoint, r.SetDesiredHPAEndpoint); err != nil {
+		return err
+	}
+
+	endpointsSpec := r.NooBaa.Spec.Endpoints
```

Comment:
What if there is no endpoints spec in the NooBaa CR?
Why not use `r.HPAEndpoint` instead?

Reply:
@guymguym Greate catch